pkg/util/limiter/memory_consumption.go

```diff
@@ -159,7 +163,14 @@ func (l *MemoryConsumptionTracker) DecreaseMemoryConsumption(b uint64, source Me
 	defer l.mtx.Unlock()
 
 	if b > l.currentEstimatedMemoryConsumptionBySource[source] {
-		panic(fmt.Sprintf("Estimated memory consumption of all instances of %s in this query is negative. This indicates something has been returned to a pool more than once, which is a bug. The affected query is: %v", source, l.queryDescription))
+		traceID, sampled := tracing.ExtractSampledTraceID(l.ctx)
+		traceDescription := ""
+
+		if sampled {
+			traceDescription = fmt.Sprintf(" (trace ID: %v)", traceID)
+		}
```

Comment:
Given you're going to panic, I think you should log the trace ID even if it wasn't sampled. You may still find that trace in some logs from other components that would help you debugging. I'd just take `trace.SpanFromContext(l.ctx).SpanContext().TraceID()`

Reply:
You can also use `tracing.ExtractTraceID()` 👍

Reply:
Changed in 7b0dcfa.